haskell/private/providers.bzl

```diff
@@ -1,3 +1,23 @@
+DefaultCompileInfo = provider(
```

Comment:
Maybe rename it to `HaskellCompileInfo`? `DefaultCompileInfo` is a bit too general.

Reply:
I agree. I just kept the initial name for now to avoid making this PR bigger than it already is.